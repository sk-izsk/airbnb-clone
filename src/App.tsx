import React, { Suspense } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { AppProvider } from './AppProvider';
import { LoadingScreen } from './components';
import { HomeScreen } from './lazyComponents';

export interface AppProps {}

const App: React.FC<AppProps> = () => {
  return (
    <AppProvider>
      <Switch>
        <Suspense fallback={<LoadingScreen />}>
          <Route exact path='/'>
            <Redirect to='/home' />
          </Route>
          <Route path='/home' exact component={HomeScreen} />
        </Suspense>
      </Switch>
    </AppProvider>
  );
};

export { App };
