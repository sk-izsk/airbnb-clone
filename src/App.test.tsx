import { render } from '@testing-library/react';
import React from 'react';
import { App } from './App';

test('App renders correctly', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
