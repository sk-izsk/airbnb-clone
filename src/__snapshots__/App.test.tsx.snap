// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-app-1"
  >
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-header-3"
    >
      <img
        alt=""
        class="makeStyles-headerIcon-4"
        src="https://i.pinimg.com/originals/3c/bf/be/3cbfbe148597341fa56f2f87ade90956.png"
      />
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-headerCenter-5"
      >
        <input
          class="makeStyles-searchField-7"
          type="text"
        />
        <svg
          fill="currentColor"
          height="20"
          size="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-headerRight-6"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Become a host
        </p>
        <svg
          fill="currentColor"
          height="20"
          size="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,23 C18.0751322,23 23,18.0751322 23,12 C23,5.92486775 18.0751322,1 12,1 C5.92486775,1 1,5.92486775 1,12 C1,18.0751322 5.92486775,23 12,23 Z M12,23 C15,23 16,18 16,12 C16,6 15,1 12,1 C9,1 8,6 8,12 C8,18 9,23 12,23 Z M2,16 L22,16 M2,8 L22,8"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
        <svg
          fill="currentColor"
          height="20"
          size="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-loaderContainer-14"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 200px; height: 200px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-13 makeStyles-footer-11"
    >
      <p
        class="MuiTypography-root makeStyles-text-12 MuiTypography-body1"
      >
        © 2020 Airbnb clone! No rights reserved - this is a demo!
      </p>
      <span
        class="MuiTypography-root makeStyles-text-12 MuiTypography-caption"
      >
        Privacy · Terms · Sitemap · Company Details
      </span>
    </div>
  </div>
</DocumentFragment>
`;
